Updated Measure-ParameterBlockParameterAttribute. Fixes #185

indented-automation opened the pull request:
* Refactored Measure-ParameterBlockParameterAttribute
* Added unit tests
* Added support for other named parameters in ParameterAttribute

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/dscresource.tests/187)
<!-- Reviewable:end -->

msftclas commented:
@__indented-automation__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

msftclas commented:
@__indented-automation__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

msftclas commented:
__@indented-automation__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

indented-automation commented:
The unit tests need script analyzer. It can either be injected into AppVeyor.psm1, or you can have a modified version of the function I use to test rules like this from my set of custom rules:

https://github.com/indented-automation/ScriptAnalyzerRules/blob/master/CodingConventions/public/Invoke-CodingConventionRule.ps1

Which would you prefer?

johlju commented:
Thanks a lot for sending in this PR! A big improvement, and I learned a lot thanks to you! :smile: 

Just minor comments

---

Reviewed 1 of 1 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[AppVeyor.psm1, line 86 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-6e4Kc9SpTtgV5gLR:-Ks-6e4Kc9SpTtgV5gLS:b-x54vib) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/AppVeyor.psm1#L86)):*
> ```PowerShell
>     }
> 
>     Install-Module @installPSScriptAnalyzerParameters
> ```

It think it's better that the test file DscResource.AnalyzerRules.Tests.ps1 call the helper function `Import-PSScriptAnalyzer`
https://github.com/PowerShell/DscResource.Tests/blob/41517cb10f8644f7228e3bac5fcdf8f75e29866e/TestHelper.psm1#L749-L785

Same function is called in Meta.Tests.ps1 when actually testing DSC resource. It then also will be possible to run the tests DscResource.AnalyzerRules.Tests.ps1 by itself. 

`Import-PSScriptAnalyzer` could be extended to do nothing if the module is already imported in the session.

What do you think?

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 37 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-7e9VmHrFAMTNBweZ:-Ks-7e9VmHrFAMTNBwe_:bvtuecb) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L37)):*
> ```PowerShell
>         [ValidateNotNullOrEmpty()]
>         [System.Management.Automation.Language.ParameterAst]
>         $ast
> ```

Please change parameter name to `$ParameterAst` (throughout)

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 71 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-88GChOAUl6S4AcI8:-Ks-88GChOAUl6S4AcI9:b-qx22eu) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L71)):*
> {
>             $args[0] -is [System.Management.Automation.Language.NamedAttributeArgumentAst] -and
>             $args[0].ArgumentName -eq 'Mandatory'
>         }

For readability. Could you please add this to a variable instead, i.e. `$mandatoryNamedArgumentFilter`.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 11 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-90uabOokmZLjNMuM:-Ks-90uabOokmZLjNMuN:bpzx1p0) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L11)):*
> 'ParameterBlockAttributeMissing'

Could you change the Context-blocks to 'When...', for example `'When parameter block attribute is missing'`. Throughout.
It lines up how most tests are written in DSC Resource Kit.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 12 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-9R5mjHYtubwHHZUY:-Ks-9R5mjHYtubwHHZUZ:bdulf46) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L12)):*
> ```PowerShell
>     Describe 'Measure-ParameterBlockParameterAttribute' {
>         Context 'ParameterBlockAttributeMissing' {
>             It 'Writes a record, when ParameterAttribute is missing' {
> ```

Could you change the It-blocks to 'Should...', for example `'Should write the correct error record'`. Throughout.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 23 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-8c_bMftTVTpATV-T:-Ks-8c_bMftTVTpATV-U:b-ajvrkw) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L23)):*
> @($record).Count

@($null).Count will always return 1 - maybe do this in another way? Throughout

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-:-Ks-9jSw8TT0zU3WqsOX:bkfq9ls)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 94 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-P4bJl1c3nosX2yl2:-Ks-P4bJl1c3nosX2yl3:br9fqxu) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L94)):*
> ```PowerShell
>             }
>             
>             if ($mandatoryNamedArgument.Argument.VariablePath.UserPath -cne 'true')
> ```

I ran a test and saw that there is no check when `Mandatory = $false` which should return `$localizedData.ParameterBlockNonMandatoryParameterMandatoryAttributeWrongFormat`.

Line 443 below is reported wrong.

Test code: 
https://github.com/johlju/xFailOverCluster/blob/fdfbd41904a6809ca574f8e402efa39eb783652f/DSCResources/MSFT_xCluster/MSFT_xCluster.psm1#L413-L456

Result: 
https://ci.appveyor.com/project/johlju/xfailovercluster/build/1.6.252.0#L330

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187)*
<!-- Sent from Reviewable.io -->

indented-automation commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 37 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-7e9VmHrFAMTNBweZ:-Ks324oofysQ8RQl41HA:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L37)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change parameter name to `$ParameterAst` (throughout)
</blockquote></details>

Done.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 71 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-88GChOAUl6S4AcI8:-Ks328B16SRoUQYzfJgi:bca1wcd) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L71)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> {
>             $args[0] -is [System.Management.Automation.Language.NamedAttributeArgumentAst] -and
>             $args[0].ArgumentName -eq 'Mandatory'
>         }

For readability. Could you please add this to a variable instead, i.e. `$mandatoryNamedArgumentFilter`.
</blockquote></details>

No longer relevant. The rules have been split and do not contain embedded AST queries.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 94 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-P4bJl1c3nosX2yl2:-Ks360BJx-mDpv-Fffaz:bcrj1v3) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L94)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I ran a test and saw that there is no check when `Mandatory = $false` which should return `$localizedData.ParameterBlockNonMandatoryParameterMandatoryAttributeWrongFormat`.

Line 443 below is reported wrong.

Test code: 
https://github.com/johlju/xFailOverCluster/blob/fdfbd41904a6809ca574f8e402efa39eb783652f/DSCResources/MSFT_xCluster/MSFT_xCluster.psm1#L413-L456

Result: 
https://ci.appveyor.com/project/johlju/xfailovercluster/build/1.6.252.0#L330
</blockquote></details>

Fixed, splitting the rules avoids use of Find (rather than FindAll) which was preventing this being discovered. I've added tests to ensure this doesn't creep back in.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 11 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-90uabOokmZLjNMuM:-Ks32JWkafQIARR9Kh6s:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L11)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> 'ParameterBlockAttributeMissing'

Could you change the Context-blocks to 'When...', for example `'When parameter block attribute is missing'`. Throughout.
It lines up how most tests are written in DSC Resource Kit.
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 12 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-9R5mjHYtubwHHZUY:-Ks32Krpx_BuzQMloKOL:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L12)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you change the It-blocks to 'Should...', for example `'Should write the correct error record'`. Throughout.
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 23 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-8c_bMftTVTpATV-T:-Ks32MjgaF8PDmB-PonY:bp2qk40) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L23)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> @($record).Count

@($null).Count will always return 1 - maybe do this in another way? Throughout
</blockquote></details>

While @($null).Count is 1, this is 0:

function test { }
$result = test
@($result).Count

However, I've replaced this mechanism with Measure-Object for clarity.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187)*
<!-- Sent from Reviewable.io -->

indented-automation commented:
Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions.

---

*[AppVeyor.psm1, line 86 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-6e4Kc9SpTtgV5gLR:-Ks37hiJEDOYzjoc7nbU:b-dgsyjp) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/AppVeyor.psm1#L86)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

It think it's better that the test file DscResource.AnalyzerRules.Tests.ps1 call the helper function `Import-PSScriptAnalyzer`
https://github.com/PowerShell/DscResource.Tests/blob/41517cb10f8644f7228e3bac5fcdf8f75e29866e/TestHelper.psm1#L749-L785

Same function is called in Meta.Tests.ps1 when actually testing DSC resource. It then also will be possible to run the tests DscResource.AnalyzerRules.Tests.ps1 by itself. 

`Import-PSScriptAnalyzer` could be extended to do nothing if the module is already imported in the session.

What do you think?
</blockquote></details>

Agreed. I hadn't noticed that function.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187)*
<!-- Sent from Reviewable.io -->

johlju commented:
Awesome work on the tests! Still an issue, and some style changes. 

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 94 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-P4bJl1c3nosX2yl2:-Ks3HT_OG2mO2Ux3D7B8:bdka6f8) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L94)):*
<details><summary><i>Previously, indented-automation (Chris Dent) wrote…</i></summary><blockquote>

Fixed, splitting the rules avoids use of Find (rather than FindAll) which was preventing this being discovered. I've added tests to ensure this doesn't creep back in.
</blockquote></details>

Sorry, but this is still wrong :smile: 

Still not seeing you output `$localizedData.ParameterBlockNonMandatoryParameterMandatoryAttributeWrongFormat` so this line 443 is hit, but it is still getting the wrong error message 
https://ci.appveyor.com/project/johlju/xfailovercluster/build/1.6.253.0#L331

https://github.com/johlju/xFailOverCluster/blob/fdfbd41904a6809ca574f8e402efa39eb783652f/DSCResources/MSFT_xCluster/MSFT_xCluster.psm1#L443

The error message should read

"Non-mandatory parameters must use the correct format [Parameter()] for the parameter attribute. See https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#correct-format-for-parameter-block"

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 1 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3INZOJQPek-nq6rcJ:-Ks3INZOJQPek-nq6rcK:b-vbpt3y) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L1)):*
> $psscriptroot

Please change to $PSScriptRoot

Please used named parameters `-Path`

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 2 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3IQXdvnEebDFbkz95:-Ks3IQXdvnEebDFbkz96:b6qgap8) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L2)):*
> $pscommandpath

Please change to $PSCommandPath

Please used named parameters`-Path`

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3JIPrZ2d73SWn234x:-Ks3JIPrZ2d73SWn234y:bfmvqff) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L3)):*
> ```PowerShell
> $script:ProjectRoot = (Get-Item $psscriptroot).Parent.Parent.FullName
> $script:ModuleName = (Get-Item $pscommandpath).BaseName -replace '\.Tests'
> $script:ModuleRootPath = Join-Path $script:ProjectRoot $script:ModuleName
> ```

Please used named parameters`-Path`, `-ChildPath`

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 7 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3JQFuGlG_pLc2f_FQ:-Ks3JQFuGlG_pLc2f_FR:b-1arc47) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L7)):*
> ```PowerShell
> Describe "$($script:ModuleName) Unit Tests" {
>     BeforeAll {
>         Import-Module -Name (Join-Path -Path $script:ProjectRoot -ChildPath "TestHelper.psm1") -Force
> ```

Please use single quotes.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 16 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3Jd23hHim7dN3vEEB:-Ks3Jd23hHim7dN3vEEC:bpwcmbl) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L16)):*
> record

Please change to the 'the correct record' because we don't expect any record, only the correct record. :) (Throughout where applicable)

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 317 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3LIigHIvzThefAZ9Y:-Ks3LIigHIvzThefAZ9Z:b-2eyctd) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L317)):*
> ```PowerShell
>                             $ParameterName1,
> 
>                             [Parameter(Mandatory = $false)]
> ```

This should have been hit by the error message not yet implemented (see comment above).

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 337 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3L8t6lwvikbjUomzc:-Ks3L8t7Rmrng3fVT3k4:b-2eyctd) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L337)):*
> ```PowerShell
>                             $ParameterName1,
> 
>                             [Parameter(mandatory = $false)]
> ```

This should have been hit by the error message not yet implemented (see comment above).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-:-Ks3LNgFhJLxK-ry-F36:b-opcte6)*
<!-- Sent from Reviewable.io -->

indented-automation commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 94 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-P4bJl1c3nosX2yl2:-Ks3TdaBE6ztGCPxn7gA:bx9dlfl) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L94)):*
> ave been hit by the error message not yet implemente
Sorry, I see what you mean now. Monday morning...

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 1 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3INZOJQPek-nq6rcJ:-Ks3Rc9sjWCqQfzfwS4g:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L1)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> $psscriptroot

Please change to $PSScriptRoot

Please used named parameters `-Path`
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 2 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3IQXdvnEebDFbkz95:-Ks3RiTiKqa6zuyMu1q8:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L2)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> $pscommandpath

Please change to $PSCommandPath

Please used named parameters`-Path`
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 3 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3JIPrZ2d73SWn234x:-Ks3RjOBR__9JkySuvMp:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L3)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please used named parameters`-Path`, `-ChildPath`
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 7 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3JQFuGlG_pLc2f_FQ:-Ks3RmvN5cS2MlMhYNsX:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L7)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please use single quotes.
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 16 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3Jd23hHim7dN3vEEB:-Ks3SDudUSTMMFSIlHvM:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L16)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> record

Please change to the 'the correct record' because we don't expect any record, only the correct record. :) (Throughout where applicable)
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 317 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3LIigHIvzThefAZ9Y:-Ks3TnK10SY2bz6Eg7UQ:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L317)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This should have been hit by the error message not yet implemented (see comment above).
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.AnalyzerRules.Tests.ps1, line 337 at r4](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks3L8t6lwvikbjUomzc:-Ks3To0CSu6cftPACdUn:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/c197b0c545a923445e4ca8a1632304a788ffee41/Tests/Unit/DscResource.AnalyzerRules.Tests.ps1#L337)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This should have been hit by the error message not yet implemented (see comment above).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187)*
<!-- Sent from Reviewable.io -->

indented-automation commented:
Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions.

---

*[DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1, line 94 at r3](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-Ks-P4bJl1c3nosX2yl2:-Ks3WnGLwuiVpfbqmmv2:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/d2fc0977e6ebf62aa80c3c9b8c2da47c050e2a22/DscResource.AnalyzerRules/DscResource.AnalyzerRules.psm1#L94)):*
<details><summary><i>Previously, indented-automation (Chris Dent) wrote…</i></summary><blockquote>

> ave been hit by the error message not yet implemente
Sorry, I see what you mean now. Monday morning...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

@kwirkykat can you look over this and merge? Returns the expected result (tested OK); https://ci.appveyor.com/project/johlju/xfailovercluster/build/1.6.254.0#L328

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresource.tests/187#-:-Ks3_ALFnIBQeoTItfMl:b-yrxjw6)*
<!-- Sent from Reviewable.io -->

johlju commented:
@kwirkykat Btw. This is a refactor of existing code, and its already entries in the unreleased section of the change log, so thought that another entry was unnecessary.
